sdk/src/Services/S3/Custom/Transfer/Internal/BufferedMultipartStream.cs

```diff
@@ -290,7 +292,13 @@ protected override void Dispose(bool disposing)
             {
                 try
                 {
-                    // Dispose modular dependencies
+                    // Dispose InitialResponse first (contains HTTP connection and network stream)
+                    // This is critical because GetObjectResponse holds unmanaged resources that
+                    // won't be cleaned up by GC - must be explicitly disposed to return HTTP
+                    // connection to the pool and close network streams
+                    _discoveryResult?.InitialResponse?.Dispose();
```

Comment:
unrelated to the optimizing stream for multi part download, but i discovered this potential resource leak while working on this